optimum/intel/neural_compressor/utils.py

```diff
@@ -91,3 +115,378 @@ def load_quantized_model(checkpoint_dir_or_file: str, model: torch.nn.Module, **
+            _, is_replaced = _replace_linear(
+                module,
+                modules_to_not_convert,
+                current_key_name,
+                quantization_config,
+                is_replaced=is_replaced,
+                device=device,
+                empty_weights=empty_weights,
+            )
+        # Remove the last key for recursion
+        current_key_name.pop(-1)
+    return model, is_replaced
+
+
+def default_run_fn(model, tokenizer, dataset, max_length=512, n_samples=100, batch_size=8, algo="rtn"):
+    from torch.utils.data import DataLoader
+
+    if isinstance(dataset, (str, bytes, os.PathLike)):
+        calib_dataset = load_dataset(dataset, split="train")
+    calib_dataset = calib_dataset.shuffle(seed=42)
+    if tokenizer is None:
+        logger.error("Please provide the tokenizer in quantization_config.")
+        exit(0)
+
+    def tokenize_function(examples):
+        if "text" in examples:
+            example = tokenizer(examples["text"])
+        else:
+            logger.error(
+                "Please check dataset prompt identifier," + " NeelNanda/pile-10k is default used calibration dataset."
+            )
+            exit(0)
+        return example
+
+    tokenized_dataset = calib_dataset.map(tokenize_function, batched=True)
+    tokenized_dataset.set_format(type="torch", columns=["input_ids"])
+    tokenized_dataset = tokenized_dataset.filter(lambda x: x["input_ids"].shape[-1] >= max_length)
+
+    def collate_batch(batch):
+        input_ids_padded = []
+        for text in batch:
+            input_ids = text["input_ids"]
+            if len(input_ids) >= max_length:
+                input_ids = input_ids[:max_length]
+                input_ids_padded.append(input_ids)
+            else:
+                continue
+        assert (
+            input_ids_padded != []
+        ), "The dataset does not have data that meets the required input length. Please reduce seq_len."
+        return torch.vstack(input_ids_padded)
+
+    calib_dataloader = DataLoader(
+        tokenized_dataset,
+        batch_size=batch_size,
+        shuffle=False,
+        collate_fn=collate_batch,
+    )
+    total_cnt = 0
+    for i, (input_ids) in enumerate(calib_dataloader):
+        if total_cnt + input_ids.shape[0] > n_samples:
+            input_ids = input_ids[: n_samples - total_cnt, ...]
+        total_cnt += input_ids.shape[0]
+        if total_cnt >= n_samples:
+            break
+
+        try:
+            model(
+                input_ids=input_ids,
+            )
+        except ValueError:
+            pass
+
+
+@torch.no_grad()
+def run_fn_for_autoround(model, dataloader):
+    for data in dataloader:
+        if isinstance(data, tuple) or isinstance(data, list):
+            model(*data)
+        elif isinstance(data, dict):
+            model(**data)
+        else:
+            model(data)
+
+
+def convert_to_weight_only_quantized_model(model, config, device="cpu"):
```

Comment:
I don't think this should be moved to optimum-intel but instead to neural-compressor (who will take care of applying the quant algo), this will prevent any compatibility issues